### DISCUSSION TITLE
Why does reload reset the key column?

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#1663] [Q&A]
I was just surprised by something. I've got a JSON file that's being periodically updated with the results of an experiment, i.e. a row is added every two minutes. I set a key column and plotted one of the columns. I pressed Ctrl-R to get the new rows, then tried to plot again, but the key column was un-keyed! Why would I have to set it again as a key column every time I reload?

**Comment by saulpw** [comment]
I think `reload` means "reload the sheet from its source as though it had just been opened", not "reload the rows but keep everything else about the sheet the same".  There could be a `reload-rows` command (`z Ctrl+R` maybe?) which does what you suggest.

**Comment by cool-RR** [comment]
I'd like that command, yes. Personally I'd bind `reload` to `gCtrl-R` so `reload-rows` could be bound to `Ctrl-R`, because I'm really afraid to lose all the changes I did. Note that `Shift-U` doesn't bring those changes back!

Also, I just noticed that for some reason, this reset behavior happens in CSV and JSON sheets, but not when using `vd` to explore a folder. Try setting some column types and reloading, it keeps the column types.

**Reply by saulpw** [reply]
It's because the columns are fixed in the DirSheet, instead of being added at load time.

**Reply by cool-RR** [reply]
Added a feature request: https://github.com/saulpw/visidata/issues/1683
